src/shared/dissect-image.c

```diff
@@ -456,6 +456,22 @@ static int device_wait_for_initialization_harder(
 
 #define DEVICE_TIMEOUT_USEC (45 * USEC_PER_SEC)
 
+static void dissected_partition_done(DissectedPartition *p) {
+        assert(p);
+
+        free(p->fstype);
```

Comment:
given the pointer is not invalidated and numeric values are reset to -1 below, wouldn't mfree to NULL these be safer?

Reply:
C gives the guarantee that on structured initialization unlisted fields are reset to zero. Thus, the initialization will set all those pointers implicitly to NULL for us.

Our general rule hence is to never specify fields during structured initialization that shall be zero/NULL/false anyway.

And thus changing this specific line to `p->fstype = mfree(p->fstype)` is something we could do, but unnecessary hence.

Reply:
Ah yes of course, there's the structured init later - duh!